### PR TITLE
Enable the `COM` and `SIM` rules in `ruff` configuration

### PR DESCRIPTION
**Pull request by Czaki**
# Description

This PR enables `COM` and `SIM` rules in `ruff` configuration. 
As we disable `COM812` the com currently is guarding for obsolete commas. 
`SIM` is a set of rules to keep code simple. We decide to disable `SIM117` as black wraps with statements in an ugly way and for a better one python 3.10+ is required.  

This rule ensures that a comma is added in the multiline function calls, dict creation etc.

The benefit of this rule is reducing diff for adding the next element (as a comma is already present), so it should reduce merge conflicts. 

## Type of change

- [x] Maintenance

# References

https://github.com/charliermarsh/ruff#flake8-commas-com

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Comment by Czaki** [comment]
Failures are unrelated. Both are in qt dims.

**Comment by brisvag** [comment]
Ouch! image

Poor github diff tool cannot handle this PR xD

A cursory glance seems to suggest that commas are added also to non-multiline calls:

```py
do_something(
	arg1, arg2, arg3, arg4, arg5,
)
```
That last comma seems unnecessary, and is probably adding extra diff here that doesn't really help?

**Comment by Carreau** [comment]
This is likely to conflict on all existing PRs. Is there some ways to introduce that progressively ?

**Comment by brisvag** [comment]
Yeah I feel like this is gonna cause more problems than the small benefit of ever-so-slighlty smaller diffs.

**Comment by Czaki** [comment]
Ok, I modify this PR to omit `COM812` rule. I like when it adds a comma on the last line of the multiline function call, but not in the single line. It should be much better now. 

Asi it becomes small, maybe I should enable more rules in this PR?

File: napari/_app_model/constants/_commands.py [hunk 1]
```diff
@@ -110,40 +110,40 @@ class _i(NamedTuple):
...
     # Help menubar
-    CommandId.NAPARI_GETTING_STARTED: _i(trans._('Getting started'), ),
-    CommandId.NAPARI_TUTORIALS: _i(trans._('Tutorials'), ),
-    CommandId.NAPARI_LAYERS_GUIDE: _i(trans._('Using Layers Guides'), ),
-    CommandId.NAPARI_EXAMPLES: _i(trans._('Examples Gallery'), ),
-    CommandId.NAPARI_RELEASE_NOTES: _i(trans._('Release Notes'), ),
-    CommandId.NAPARI_HOMEPAGE: _i(trans._('napari homepage'), ),
-    CommandId.NAPARI_INFO: _i(trans._('napari Info'), ),
-    CommandId.NAPARI_GITHUB_ISSUE: _i(trans._('Report an issue on GitHub'), ),
-    CommandId.TOGGLE_BUG_REPORT_OPT_IN: _i(trans._('Bug Reporting Opt In/Out...'), ),
```
Comment (brisvag) [inline napari/_app_model/constants/_commands.py:121]: would be nice to also delete the spaces here, I guess :P

**Review by brisvag — COMMENTED** [review]
Yeah, I'm ok with this. Feel free to add more rules if you think they make sense!

File: napari/_app_model/constants/_commands.py (hunk 1 above)
Comment (Czaki) [reply napari/_app_model/constants/_commands.py:121]: fixed

**Comment by Czaki** [comment]
Maybe some merge of `with` and `if` are not looking bests, but I like the idea as it could reveal some potential problems caused by missing some branches. 

I think that it is enough big for review.

File: napari/_qt/_tests/test_plugin_widgets.py [hunk 2]
```diff
@@ -192,6 +192,7 @@ def test_inject_viewer_proxy(make_napari_viewer):
     assert isinstance(wdg.viewer, PublicOnlyProxy)
 
     # simulate access from outside napari
-    with patch('napari.utils.misc.ROOT_DIR', new='/some/other/package'):
-        with pytest.warns(FutureWarning):
-            wdg.fail()
+    with patch(
+        'napari.utils.misc.ROOT_DIR', new='/some/other/package'
+    ), pytest.warns(FutureWarning):
+        wdg.fail()
```
Comment (brisvag) [inline napari/_qt/_tests/test_plugin_widgets.py]: this is black doing its thing? ugh... Is it possible to get this instead?

```py
    with (
		patch('napari.utils.misc.ROOT_DIR', new='/some/other/package'),
		pytest.warns(FutureWarning),
	):
        wdg.fail()
```

**Review by brisvag — COMMENTED** [review]
Generally like the idea, but man those newlines are unreadable sometimes...

File: napari/_qt/_tests/test_plugin_widgets.py (hunk 2 above)
Comment (Czaki) [reply napari/_qt/_tests/test_plugin_widgets.py]: It is black and 80 character limits. I will try to work around this.

Comment (Czaki) [reply napari/_qt/_tests/test_plugin_widgets.py]: The code that you suggest is available from python 3.10. We need to drop 3.8 and 3.9 to use it...

Comment (brisvag) [reply napari/_qt/_tests/test_plugin_widgets.py]: ^^' I vote in favour!

---

Jokes apart, I find the nested `with` way more readable than this, so I think I would rather disable this rule for now.

Comment (Czaki) [reply napari/_qt/_tests/test_plugin_widgets.py]: I have reverted part of `SIM117` changes (keep this good-looking) and disabled this rule.

File: napari/components/viewer_model.py [hunk 3]
```diff
@@ -1052,7 +1052,7 @@ def open(
         ) as pbr:
             for _path in pbr:
                 # If _path is a list, set stack to True
-                _stack = True if isinstance(_path, list) else False
+                _stack = bool(isinstance(_path, list))
```
Comment (brisvag) [inline napari/components/viewer_model.py]:
```suggestion
                _stack = isinstance(_path, list)
```

File: napari/layers/points/points.py [hunk 4]
```diff
@@ -521,75 +521,65 @@ def data(self, data: Optional[np.ndarray]):
...
-                        size = np.repeat([new_size], adding, axis=0)
-
-                        if len(self._edge_width) > 0:
-                            new_edge_width = copy(self._edge_width[-1])
-                        else:
-                            new_edge_width = self.current_edge_width
-                        edge_width = np.repeat(
-                            [new_edge_width], adding, axis=0
-                        )
+        with self.events.blocker_all(), self._edge.events.blocker_all(), self._face.events.blocker_all():
```
Comment (brisvag) [inline napari/layers/points/points.py:524]: this is too long right?

File: napari/utils/misc.py [hunk 5]
```diff
@@ -235,17 +232,18 @@ def ensure_sequence_of_iterables(
...
-                raise ValueError(
-                    trans._(
-                        "length of {obj} must equal {length}",
-                        deferred=True,
-                        obj=obj,
-                        length=length,
-                    )
+        if (length is not None and len(obj) != length) and (
+            (len(obj) == 0 and not repeat_empty) or len(obj) > 0
+        ):
```
Comment (brisvag) [inline napari/utils/misc.py]: Would prefer this to be in separate lines for each thing "at the same level"

File: napari/layers/image/experimental/_octree_loader.py [hunk 6]
```diff
@@ -189,9 +189,9 @@ def get_drawable_chunks(
             # The ideal level is priority 0, 1 is one level above idea, etc.
             priority = chunk.location.level_index - ideal_level
 
-            if chunk.in_memory:
-                drawable.append(chunk)
-            elif chunk.needs_load and self._load_chunk(chunk, priority):
+            if chunk.in_memory or (
+                chunk.needs_load and self._load_chunk(chunk, priority)
+            ):
```
Comment (brisvag) [inline napari/layers/image/experimental/_octree_loader.py]: Would prefer this to be in separate lines for each thing "at the same level".

File: napari/utils/geometry.py [hunk 7]
```diff
@@ -695,11 +692,10 @@ def find_front_back_face(
...
                 front_face_normal = v
-        elif (np.dot(view_dir, v) + 0.001) > 0:
-            if line_in_quadrilateral_3d(
-                click_pos, view_dir, bbox_face_coords[k]
-            ):
-                back_face_normal = v
+        elif (np.dot(view_dir, v) + 0.001) > 0 and line_in_quadrilateral_3d(
+            click_pos, view_dir, bbox_face_coords[k]
+        ):
+            back_face_normal = v
```
Comment (brisvag) [inline napari/utils/geometry.py]: Also here

**Review by brisvag — COMMENTED** [review]
With very few exceptions, every change in here is an improvement in readability. Love it!

File: napari/layers/points/points.py (hunk 4 above)
Comment (Czaki) [reply napari/layers/points/points.py:524]: It is, I do not understand why black do not touch it.

**Comment by Czaki** [comment]
@brisvag I have rewritten the logic of these three problematic places. Could you check it?

File: napari/layers/image/experimental/_octree_loader.py [hunk 8]
```diff
@@ -189,9 +189,10 @@ def get_drawable_chunks(
             # The ideal level is priority 0, 1 is one level above idea, etc.
             priority = chunk.location.level_index - ideal_level
 
-            if chunk.in_memory or (
-                chunk.needs_load and self._load_chunk(chunk, priority)
-            ):
+            if chunk.needs_load:
+                self._load_chunk(chunk, priority)
+
+            if chunk.in_memory:
```
Comment (brisvag) [inline napari/layers/image/experimental/_octree_loader.py:195]: Wow, this was very convoluted before o.o

**Review by brisvag — APPROVED** [review]
I think it makes sense... These if statements are a bit hard to parse. Hopefully the tests save us!

File: napari/utils/misc.py [hunk 9]
```diff
@@ -234,18 +231,18 @@ def ensure_sequence_of_iterables(
...
+                trans._(
+                    "length of {obj} must equal {length}",
+                    deferred=True,
+                    obj=obj,
+                    length=length,
                 )
+            )
 
         if len(obj) > 0 or not repeat_empty:
             return obj
```
Comment (Czaki) [inline napari/utils/misc.py:248]: In general this function looks problematic that implementation do not cover `repeat_empty` with `length != None` and it is not mentioned in the docs

Comment (brisvag) [reply napari/utils/misc.py:248]: I see... for now it's ok to leave it as as, or you can add that branch if you like, but I wouldn't worry about that in this PR.

**Review by jni — APPROVED** [review]
Nice! @Czaki can you add the SIM rule also to the description of the PR so that it is described in the commit?

**Comment by Czaki** [comment]
@jni could you check now?

**Comment by jni** [comment]
Ready to go as far as I'm concerned!
